ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -30,22 +35,100 @@ const styles = reactStyles({
   narrowWithMargin: {
     width: '10rem',
     marginRight: '1rem'
-  }
-});
+  },
+  fileDetailsTable: {
+    maxWidth: '1000px',
+    marginTop: '1rem',
+  },
+})
 
 const PurpleLabel = ({style = {}, children}) => {
   return <label style={{color: colors.primary, ...style}}>
     {...children}
   </label>;
 };
 
+const workspaceInfoField = (labelText: string, divContents: React.ReactFragment) => {
+  return <FlexRow style={{width: '80%', maxWidth: '1000px'}}>
+    <PurpleLabel
+        style={{
+          width: '250px',
+          minWidth: '180px',
+          textAlign: 'right',
+          marginRight: '1rem',
+        }}
+    >
+      {labelText}
+    </PurpleLabel>
+    <div
+        style={{
+          flex: 1,
+          wordWrap: 'break-word',
+        }}
+    >
+      {divContents}
+    </div>
+  </FlexRow>;
+};
+
+const FileDetails = (props: {data: WorkspaceFiles}) => {
+  const {data} = props;
+  const {bucket} = data;
+
+  interface TableEntry {
+    name: string;
+    size: number;
+    location: string;
+  }
+
+  const getLocation = (file: FileDetail): string => {
```

Comment:
alt: parseLocation, deriveLocation, etc?

Reply:
parseLocation seems reasonable to me